duckduckgo/Android
# New RMF message type: Promo Single Action

Task/Issue URL: https://app.asana.com/0/488551667048375/1204954301997720/f

### Description
Implemented new type of remote message called Promo Single Action

### Steps to test this PR
ℹ️ Change endpoint in `RemoteMessagingService.kt` to `https://www.jsonblob.com/api/1126823241334407168`

_New Promo Single Action Remote Message_
- Fresh install
- Skip onboarding
- [ ] Check you see new Promo Single Action message
- [ ] Check you see new MacAndWindows illustration
- [ ] Check "duckduckgo.com/browser" in the description is bold
- [ ] Check you see new action button with an icon

_New Share Action_
- Fresh install
- Skip onboarding
- Tap on "Share Link" button
- [ ] Check share sheet appears
- Dismiss the share sheet
- [ ] Check Remote message is **not** dismiss
- Tap on "Share Link" button again
- Select an app to share link and go back to the app
- [ ] Check remote message is gone

_Share sheet title (optional)_
ℹ️ An optional share sheet title has been added in the JSON. However, not all devices will show it
- Fresh install
- Skip onboarding
- Tap on "Share Link" button
- [ ] Check you see "Share Title" in the share sheet

_New RMF pixels_
- Fresh install
- Skip onboarding
- Tap on "Share Link" button
- [ ] Check in _Logcat_ that pixel `m_remote_message_action_clicked` is sent
- Select an app to share the link
- [ ] Check in _Logcat_ that pixel `m_remote_message_share` is sent. 
- [ ] Check pixel parameter `success` is "true"


### UI changes
| Light  | Dark |
| ------ | ----- |
![RMF_promo_message_light](https://github.com/duckduckgo/Android/assets/20798495/ef0f8ea6-0435-48f4-8e43-83bc1c0dbced)|![RMF_promo_message_dark](https://github.com/duckduckgo/Android/assets/20798495/790b4141-1d87-4513-9fae-d16046609cc5)|
